src/providers/VoiceFocusProvider/index.tsx

```diff
@@ -0,0 +1,131 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useState, useEffect, useMemo, useContext, createContext, useCallback } from 'react';
+import {
+  Device,
+  VoiceFocusTransformDevice,
+  VoiceFocusDeviceTransformer,
+  VoiceFocusSpec,
+  VoiceFocusDeviceOptions,
+} from 'amazon-chime-sdk-js';
+
+interface Props {
+  /** Determines how you want the Amazon Voice Focus to behave. This spec is used to derive a runtime configuration when a transformer is created. */
+  spec?: VoiceFocusSpec;
+  /**
+  * A set of options that can be supplied when creating an Amazon Voice Focus device.
+  * For more info, you can go to https://aws.github.io/amazon-chime-sdk-js/interfaces/voicefocusdeviceoptions.html
+  */
+  options?: VoiceFocusDeviceOptions;
+}
+
+interface VoiceFocusState {
+  isVoiceFocusSupported: boolean;
+  addVoiceFocus: (device: Device) => Promise<Device | VoiceFocusTransformDevice>;
+}
+
+const VoiceFocusContext = createContext<VoiceFocusState | null>(null);
+
+const VoiceFocusProvider: React.FC<Props> = ({
+  spec,
+  options,
+  children
+}) => {
+  const [isVoiceFocusSupported, setIsVoiceFocusSupported] = useState(false);
+  const [voiceFocusDevice, setVoiceFocusDevice] = useState<VoiceFocusTransformDevice | null>(null);
+  const [voiceFocusTransformer, setVoiceFocusTransformer] = useState<VoiceFocusDeviceTransformer | null>(null);
+
+  useEffect(() => {
+    async function initVoiceFocus() {
+      try {
+        const isVoiceFocusSupported = await VoiceFocusDeviceTransformer.isSupported(spec, options);
+        if (isVoiceFocusSupported) {
```

Comment:
You only need to do this check once, not every time the dependency set changes.

Reply:
The different setting of `spec` and `options` won't change the value of `isVoiceFocusSupported`?

Reply:
Not for this static support check, no. It _will_ affect the return value of `specificTransformer.isSupported()`.

Reply:
Slight correction: the spec dictates the worker URL, which means that it affects whether the worker check passes or fails. Suggest simply removing this lightweight check altogether; it's not a good match for the way this component works.

Reply:
Based on our discussion, removed this if statement.